src/ode_solver/explicit_ode_solver.cpp

```diff
@@ -13,90 +13,43 @@ void ExplicitODESolver<dim,real,MeshType>::step_in_time (real dt, const bool pse
-        //this->rk_stage[1].add(dt,this->solution_update);
-        if (pseudotime) {
-            const double CFL = dt;
-            this->dg->time_scale_solution_update( this->solution_update, CFL );
-            this->rk_stage[1].add(1.0,this->solution_update);
-        } else {
-            this->rk_stage[1].add(dt,this->solution_update);
-        }
-
-        // Stage 2
-        if ((ode_param.ode_output) == Parameters::OutputEnum::verbose) {
-            this->pcout<< "2... " << std::flush;
-        }
-        this->dg->solution = this->rk_stage[1];
-        this->dg->assemble_residual ();
-        this->dg->global_inverse_mass_matrix.vmult(this->solution_update, this->dg->right_hand_side);
-
-        this->rk_stage[2] = this->rk_stage[0];
-        this->rk_stage[2] *= 0.75;
-        this->rk_stage[2].add(0.25, this->rk_stage[1]);
-        //this->rk_stage[2].add(0.25*dt, this->solution_update);
-        if (pseudotime) {
-            const double CFL = 0.25*dt;
-            this->dg->time_scale_solution_update( this->solution_update, CFL );
-            this->rk_stage[2].add(1.0,this->solution_update);
-        } else {
-            this->rk_stage[2].add(0.25*dt,this->solution_update);
-        }
-
-        // Stage 3
-        if ((ode_param.ode_output) == Parameters::OutputEnum::verbose) {
-            this->pcout<< "3... " << std::flush;
-        }
-        this->dg->solution = this->rk_stage[2];
-        this->dg->assemble_residual ();
-        this->dg->global_inverse_mass_matrix.vmult(this->solution_update, this->dg->right_hand_side);
-
-        this->rk_stage[3] = this->rk_stage[0];
-        this->rk_stage[3] *= 1.0/3.0;
-        this->rk_stage[3].add(2.0/3.0, this->rk_stage[2]);
-        //this->rk_stage[3].add(2.0/3.0*dt, this->solution_update);
-        if (pseudotime) {
-            const double CFL = (2.0/3.0)*dt;
-            this->dg->time_scale_solution_update( this->solution_update, CFL );
-            this->rk_stage[3].add(1.0,this->solution_update);
-        } else {
-            this->rk_stage[3].add((2.0/3.0)*dt,this->solution_update);
-        }
-
-        this->dg->solution = this->rk_stage[3];
-        if ((ode_param.ode_output) == Parameters::OutputEnum::verbose) {
-            this->pcout<< "done." << std::endl;
-        }
+   
+    
+    //calculating stages
+    this->solution_update = this->dg->solution; //u_ni
+    for (int i = 0; i < rk_order; ++i){
+        this->rk_stage[i] = this -> solution_update; //u_n
+        for (int j = 0; j < i; ++j){
+            if (this->butcher_a[i][j] != 0){
+                if (pseudotime) {
+                    //implemented but not tested 
+                    //to my knowledge, there aren't any existing tests using explicit steady-state 
+                    //(searched through unit_tests folder)
+                    std::cout << "Explicit pseudotime not tested!!" << std::endl;
+                    const double CFL =this-> butcher_a[i][j] * dt;
```

Comment:
```suggestion
                    const double CFL =this->butcher_a[i][j] * dt;
```

Reply:
Thanks for the formatting suggestions where I've been inconsistent!